util/chunk/codec.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chunk
+
+import (
+	"encoding/binary"
+)
+
+func Encode(chk *Chunk) []byte {
+	buffer := make([]byte, 0, chk.MemoryUsage())
+	for _, col := range chk.columns {
+		if col.isFixed() {
+			buffer = encodeFixLenColumn(buffer, col)
+		} else {
+			buffer = encodeVarLenColumn(buffer, col)
+		}
+	}
+	return buffer
+}
+
+func encodeColumn(buffer, lenBuffer []byte, col *column) []byte {
+	// encode length
+	binary.BigEndian.PutUint32(lenBuffer, uint32(col.length))
+	buffer = append(buffer, lenBuffer[:4]...)
+
+	// encode nullCount
+	binary.BigEndian.PutUint32(lenBuffer, uint32(col.nullCount))
+	buffer = append(buffer, lenBuffer[:4]...)
+
+	// encode nullBitmap
+	binary.BigEndian.PutUint32(lenBuffer, uint32(len(col.nullBitmap)))
+	buffer = append(buffer, lenBuffer[:4]...)
+	buffer = append(buffer, col.nullBitmap...)
```

Comment:
We need padding to 64 bytes, according apache arrow spec 
https://arrow.apache.org/docs/memory_layout.html